File: 1/captcha.py
import os

def part1():
	captcha = get_input('input.txt').strip()

	s = 0
	length = len(captcha)
	for i, char in enumerate(captcha):
		if char == captcha[(i + 1) % length]:
			s += int(char)

	print('Part 1: {}'.format(s))

def part2():
	captcha = get_input('input.txt').strip()

	s = 0
	length = len(captcha)
	for i, char in enumerate(captcha):
		if char == captcha[(i + int(length / 2)) % length]:
			s += int(char)

	print('Part 2: {}'.format(s))

def get_input(filename):
	filepath = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'input', filename)
	with open(filepath) as infile:
		return infile.read()

if __name__ == '__main__':
	part1()
	part2()

File: 1/captcha.rs
use std::fs::File;
use std::io::Read;

fn part1() {
	let nums = get_nums(get_input());

	let mut sum = 0;

	for i in 0..nums.len() {
		if nums[i] == nums[(i + 1) % nums.len()] {
			sum += nums[i]
		}
	}

	println!("Part 1: {}", sum);
}

fn part2() {
	let nums = get_nums(get_input());

	let mut sum = 0;

	let size = nums.len();

	for i in 0..size {
		if nums[i] == nums[(i + (size / 2)) % size] {
			sum += nums[i]
		}
	}

	println!("Part 2: {}", sum);
}

fn get_nums(s: String) -> Vec<u32> {
	s.trim().bytes()
		.map(|b| { String::from_utf8(vec![b]).unwrap() })
		.map(|s| { s.parse::<u32>().unwrap() })
		.collect()
}

fn get_input() -> String {
	let mut file = File::open("./input/input.txt").unwrap();
	let mut contents = String::new();
	file.read_to_string(&mut contents).unwrap();

	return contents;
}

fn main() {
	part1();
	part2();
}
